flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/subscriber/impl/TopicPatternSubscriber.java

```diff
@@ -52,11 +62,80 @@ public TopicPatternSubscriber(Pattern topicPattern, RegexSubscriptionMode subscr
         TopicName destination = TopicName.get(topicPattern.pattern());
         NamespaceName namespaceName = destination.getNamespaceObject();
         this.namespace = namespaceName.toString();
+        this.subscriptionMode = convertRegexSubscriptionMode(subscriptionMode);
+        this.useBinaryProtocol = true;
     }
 
     @Override
     public Set<TopicPartition> getSubscribedTopicPartitions(
             RangeGenerator generator, int parallelism) throws Exception {
+        Set<String> topics;
+        if (useBinaryProtocol) {
+            try {
+                topics = queryTopicsByInternalProtocols();
```

Comment:
Can the current test case use the new protocol? If not, it needs to be covered. In addition, it may be necessary to test the case that new protocol is not compatible while fallback to the old protocol.

Reply:
This should be a mistake from my side. After testing the new protocol with old Pulsar broker. It's binary compatible. I add this comment weeks ago because I didn't find the bug from Pulsar side which can't filter some regex.

Reply:
BTW. The new protocol has been covered by the existing tests. No need to add extra tests I think.

Reply:
Fair enough.